kubeflow/core/setup_iap.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+#
+# A simple shell script to enable IAP and configure timeouts by using gcloud.
+[ -z ${CLIENT_ID} ] && echo Error CLIENT_ID must be set && exit 1
+[ -z ${CLIENT_SECRET} ] && echo Error CLIENT_SECRET must be set && exit 1
+[ -z ${NAMESPACE} ] && echo Error NAMESPACE must be set && exit 1
+[ -z ${SERVICE} ] && echo Error SERVICE must be set && exit 1
+
+apk add --update jq
+curl https://storage.googleapis.com/kubernetes-release/release/v1.9.4/bin/linux/amd64/kubectl > /usr/local/bin/kubectl && chmod +x /usr/local/bin/kubectl
+
+# Stagger init of replicas when acquiring lock
+sleep $(( $RANDOM % 5 + 1 ))
+
+# We acquire a lock because we want to ensure  there is a single process
+# trying to modify the backend at a time.
+kubectl get svc ${SERVICE} -o json > service.json
+LOCK=$(jq -r ".metadata.annotations.iaplock" service.json)
+
+NOW=$(date -u +'%s')
+if [[ -z "${LOCK}" || "${LOCK}" == "null" ]]; then
```

Comment:
why we still need lock here if replicas=1?

Reply:
Because there could still be more than 1 pod at a given time; K8s doesn't guarantee that there is a single pod all the time; it just guarantees that in steady state there will be 1 pod.